tests/frame_submission.cpp

```diff
@@ -98,3 +99,48 @@ TEST_F(FrameSubmission, test_buffer_can_be_deleted_after_attached)
     // Roundtrip to ensure the server has processed our weirdness
     client.roundtrip();
 }
+
+/* Firefox has a lovely habit of sending an endless stream of
+ * wl_surface.frame() requests (maybe it's trying to estimate vsync?!)
+ *
+ * If a compositor responds immediately to the frame on commit, Firefox
+ * ends up looping endlessly. If the compositor *doesn't* respond to the frame
+ * request, Firefox decides not to draw anything. 🤷‍♀️
+ */
+TEST_F(FrameSubmission, test_frame_callback_storm)
```

Comment:
I don't think this test name tells us much:

    [  FAILED  ] VirtualPointerV1Test.test_frame_callback_storm

```suggestion
TEST_F(FrameSubmission, when_client_endlessly_requests_frame_then_callbacks_are_throttled)
```